src/actions/apply_action.rs

```diff
@@ -122,10 +123,56 @@ fn apply_deterministic_action(state: &mut State, action: &Action) {
         } => {
             apply_healing(action.actor, state, *in_play_idx, *amount);
         }
+        SimpleAction::SearchDeck => {
+            // Logic for searching the deck for a Pokémon card
+            apply_search_deck(action.actor, state);
+        }
         _ => panic!("Deterministic Action expected"),
     }
 }
 
+/// Function to handle searching the deck for a Grass Pokémon card (Caterpie's "Find a Friend" attack)
+fn apply_search_deck(acting_player: usize, state: &mut State) {
+    use crate::types::{Card, EnergyType};
+    use rand::{thread_rng, Rng};
+    
+    // Find all Grass-type Pokémon cards in the deck
+    let grass_pokemon_cards = state
+        .decks[acting_player]
+        .iter()
+        .enumerate()
+        .filter_map(|(idx, card)| {
+            if let Card::Pokemon(_) = card {
+                // Check if the Pokémon is Grass type
+                if card.get_type() == Some(EnergyType::Grass) {
+                    Some(idx)
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<usize>>();
+
+    if grass_pokemon_cards.is_empty() {
+        // No Grass Pokémon cards in the deck, nothing to do
+        return;
+    }
```

Comment:
Add a comment noting that we hard-code Grass type for now, but we will want to use this same function for other type of search decks.